compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -375,23 +345,23 @@ fn nameize<I: Iterator<Item = NamedMatch>>(
         res: &mut I,
         ret_val: &mut FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
     ) -> Result<(), (rustc_span::Span, String)> {
-        match *m {
-            TokenTree::Sequence(_, ref seq) => {
+        match m {
+            TokenTree::Sequence(_, seq) => {
                 for next_m in &seq.tts {
                     n_rec(sess, next_m, res.by_ref(), ret_val)?
                 }
             }
-            TokenTree::Delimited(_, ref delim) => {
-                for next_m in &delim.tts {
+            TokenTree::Delimited(_, delim) => {
+                for next_m in delim.inner_tts() {
                     n_rec(sess, next_m, res.by_ref(), ret_val)?;
                 }
             }
-            TokenTree::MetaVarDecl(span, _, None) => {
+            &TokenTree::MetaVarDecl(span, _, None) => {
```

Comment:
```suggestion
            TokenTree::MetaVarDecl(span, _, None) => {
```
Style nit: this is probably the most awfully looking pattern in patterns, especially if some arms have it and some not, and I'm not even sure that it disables the "match ergonomics" in the nested patterns.